### ISSUE TITLE
contextlib.suppress unrecognized for handling TypedDict errors

### ISSUE THREAD

**Issue by GBeauregard** [microsoft/pyright#2378]
The `TypedDict` handling here correctly determines there may be a runtime exception, but it doesn't recognize my use of `contextlib.suppress` to handle it. This may be a common enough pattern that it's worth detecting?

https://github.com/GBeauregard/pyffstream/blob/bca63b3aee23700cc8678650febd6803e7867f1f/src/pyffstream/ffmpeg.py#L360

The project currently satisfies `mpy --strict`, and I'm now working on satisfying pyright.

**Describe the bug**
pyright doesn't recognize `contextlib.suppress` to handle errors. Replacing `contextlib.suppress(KeyError)` with `try`/`except KeyError` satisfies pyright.

**To Reproduce**
pyright can be run on the above project src directory with default settings to get:
```console
  /home/gregory/Documents/projects/pyffstream/src/pyffstream/ffmpeg.py:361:25 - error: Could not access item in TypedDict
    "program_version" is not a required key in "FFProbeJSON", so access may result in runtime exception (reportTypedDictNotRequiredAccess)
  /home/gregory/Documents/projects/pyffstream/src/pyffstream/ffmpeg.py:363:24 - error: Could not access item in TypedDict
    "program_version" is not a required key in "FFProbeJSON", so access may result in runtime exception (reportTypedDictNotRequiredAccess)
```

**Expected behavior**
It should recognize the suppress context manager.

**VS Code extension or command-line**
pyright 1.1.175 being run via commandline in Linux

**Additional context**
Just using `try/except` isn't the worst thing in the world, but with suppress being in stdlib this is probably common enough to special case?

**Comment by erictraut** [comment]
Using `try/except` directly would still result in a the same error message. Pyright assumes that exceptions are used for exceptional cases, not for expected control flow. As such, it doesn't suppress type errors within `try` blocks or within context managers that handle exceptions.

**Comment by GBeauregard** [comment]
> Using `try/except` directly would still result in a the same error message. Pyright assumes that exceptions are used for exceptional cases, not for expected control flow. As such, it doesn't suppress type errors within `try` blocks or within context managers that handle exceptions.

I'm willing to accept it's poor coding practice, but there's still something funky here: I assumed this was a bug _because_ using `try/except` didn't result in the same error message. If it's as designed then there's probably a bug there instead.

**Comment by erictraut** [comment]
Ah, good point. I stand corrected in my previous statement. We do exempt this specific check within a try block. For consistency, I agree that we should treat context managers the same in this case.

**Comment by erictraut** [comment]
This is included in pyright 1.1.176, which I just published. It will also be included in the next release of pylance.
